### DISCUSSION TITLE
Prevent wire:navigate from scrolling to top after navigated?

### DISCUSSION THREAD

**Discussion by pronode** [livewire/livewire#7462] [1. Help] [answered]
I would like to prevent wire:navigate from automatically scrolling to the top of the page after navigation is complete. 

Is it possible to disable that default behavior?

**Comment by pronode** [comment, marked as answer]
I found the lines responsible for scrolling:
https://github.com/livewire/livewire/blob/main/js/plugins/navigate/scroll.js
```
export function restoreScrollPositionOrScrollToTop() {
    let scroll = el => {
        if (! el.hasAttribute('data-scroll-x')) {
            window.scrollTo({ top: 0, left: 0, behavior: 'instant' })
            ...
```
So we can register following listener:
```
document.addEventListener("livewire:navigated", () => {
    // Add data-scroll-x to body:
    document.body.setAttribute("data-scroll-x", window.scrollX);
})
```
Now Livewire kinda "maintains" scroll position.
It would be much better if this was configurable because depending on how much delay is before livewire:navigated event and actual scrolling the user may encount interface jumps.

**Comment by olssonm** [comment]
Check the documentation: https://livewire.laravel.com/docs/navigate#preserving-scroll-position 

You can preserve the scroll-position via:

```
@persist('scrollbar')
  <div class="overflow-y-scroll" wire:scroll> 
      <!-- ... -->
  </div>
@endpersist
```

**Reply by pronode** [reply]
Not the case.
